### DISCUSSION TITLE
Pragma Statement Warning on New Hardhat Installation

### DISCUSSION THREAD

**Discussion by JF0001** [NomicFoundation/hardhat#5201] [Troubleshooting]
I have completed a new installation of Hardhat. 
In the hardhat.config.js file, I have this code:
`module.exports = {
  solidity: "0.8.24",
};`

While in the default Lock.sol contract I have this pragma statement:
`pragma solidity ^0.8.24;`

However, I am getting the following warning on my pragma statement:
"Source file requires different compiler version (current compiler is 0.8.19+commit.7dd6d404.Emscripten.clang) - note that nightly builds are considered to be strictly less than the released version(5333)"

**Comment by schaable** [comment]
Closing this discussion since it has been submitted as an issue. See [Issue #5202](https://github.com/NomicFoundation/hardhat/issues/5202) for more details.
